pkg/stub/handler.go

```diff
@@ -104,6 +107,99 @@ type Handler struct {
 	mutex *sync.Mutex
 }
 
+func (h *Handler) StatusUpdate(condition v1alpha1.SamplesResourceConditionType, srcfg *v1alpha1.SamplesResource) error {
+	if condition == v1alpha1.SamplesExist {
+		cm, err := h.configmapclientwrapper.Get(h.namespace, v1alpha1.SamplesResourceName)
+		if err != nil && !kerrors.IsNotFound(err) {
+			return h.processError(srcfg, v1alpha1.SamplesUpdateFailed, err, "unable to get configm map: %v")
+		}
+		if cm == nil {
+			cm := corev1.ConfigMap{}
+			cm.Name = v1alpha1.SamplesResourceName
+			cm.Data = map[string]string{}
+			if len(srcfg.Spec.InstallType) == 0 {
+				cm.Data[installtypekey] = string(v1alpha1.CentosSamplesDistribution)
+			} else {
+				cm.Data[installtypekey] = string(srcfg.Spec.InstallType)
+			}
+			if len(srcfg.Spec.Architectures) == 0 {
+				cm.Data[x86] = x86
+			} else {
+				for _, arch := range srcfg.Spec.Architectures {
+					switch arch {
+					case x86:
+						cm.Data[x86] = x86
+					case ppc:
+						cm.Data[ppc] = ppc
+					}
+				}
+			}
+			_, err = h.configmapclientwrapper.Create(h.namespace, &cm)
+			if err != nil {
+				return h.processError(srcfg, v1alpha1.SamplesUpdateFailed, err, "problem creating configmap: %v")
+			}
+			return nil
+		}
+		err = fmt.Errorf("should never update status config map after initial creation: %#v", srcfg)
+		return h.processError(srcfg, v1alpha1.SamplesUpdateFailed, err, "%v")
+	}
+	return nil
+}
+
+func (h *Handler) SpecValidation(srcfg *v1alpha1.SamplesResource) error {
+	// if we have not had a valid SamplesResource processed, allow caller to try with
+	// the srcfg contents
+	if h.samplesResource == nil || !h.samplesResource.ConditionTrue(v1alpha1.SamplesExist) {
+		return nil
+	}
+	cm, err := h.configmapclientwrapper.Get(h.namespace, v1alpha1.SamplesResourceName)
+	if cm == nil || err != nil {
+		return h.processError(srcfg, v1alpha1.SamplesUpdateFailed, err, "problem getting configmap: %v")
+	}
+
+	switch srcfg.Spec.InstallType {
+	case "":
+		fallthrough
+	case v1alpha1.CentosSamplesDistribution:
+		installtype, ok := cm.Data[installtypekey]
+		if !ok || installtype != string(v1alpha1.CentosSamplesDistribution) {
+			err = fmt.Errorf("cannot change installtype from %s to %s", installtype, srcfg.Spec.InstallType)
+			return h.processError(srcfg, v1alpha1.SamplesUpdateFailed, err, "%v")
```

Comment:
what status conditions will this set?  i think it's going to set SamplesExist==false which is not (necessarily) true.

The samples still exist (we didn't delete anything that was previously created).


Same for the other places you call this.

We need to distinguish between "your current spec is being rejected" and "the current state of what the operator has done"

Reply:
yeah I debated that ... so to be clear, to me, that says a new condition ... an error one, perhaps called "ValidationSucceeded" or "ValidationFailed"

please confirm